python/lsst/rubintv/production/rubinTv.py

```diff
@@ -825,3 +850,201 @@ def beat(self, ensure=False, customFlatlinePeriod=None):
+
+            outputDict = {
+                '5-sigma source count': len(calibrateRes.outputCat),
+                'PSF FWHM': summaryStats.psfSigma * SIGMA2FWHM * pixToArcseconds,
+                'PSF e1': e1,
+                'PSF e2': e2,
+                'Sky mean': summaryStats.skyBg,
+                'Sky RMS': summaryStats.skyNoise,
+                'Variance plane mean': summaryStats.meanVar,
+                'PSF star count': summaryStats.nPsfStar,
+                'Astrometric bias': summaryStats.astromOffsetMean,
+                'Astrometric scatter': summaryStats.astromOffsetStd,
+                'Zeropoint': summaryStats.zeroPoint
+            }
+
+            # flag all these as measured items to color the cell
+            labels = {"_" + k: "measured" for k in outputDict.keys()}
+            outputDict.update(labels)
+
+            mdDict = {seqNum: outputDict}
+            writeMetadataShard(self.shardsDir, dayObs, mdDict)
+            self.log.info(f'Wrote metadata shard. Putting calexp. {dataId}')
+            self.safeWrite(calibrateRes.outputExposure, "calexp", dataId)
+            tFinal = time.time()
+            self.log.info(f"Ran everything in {tFinal-tStart:.2f} seconds")
+
+        except Exception as e:
+            if self.doRaise:
+                raise RuntimeError(f"Error processing {dataId}") from e
+            self.log.warning(f"Did not finish calibration of {dataId} because {repr(e)}")
+            return None
+
+    def defineVisit(self, dataId):
+        """Define a visit in the registry, given a dataId.
+
+        Note that this takes about 9ms regardless of whether it exists, so it
+        is no quicker to check than just run the define call.
+
+        NB: butler must be writeable for this to work.
+        """
+        instr = Instrument.from_string(self.butler.registry.defaults.dataId['instrument'],
+                                       self.butler.registry)
+        config = DefineVisitsConfig()
+        instr.applyConfigOverrides(DefineVisitsTask._DefaultName, config)
+
+        task = DefineVisitsTask(config=config, butler=self.butler)
+
+        task.run([butlerUtils.getExpIdFromDayObsSeqNum(self.butler, dataId)],
+                 collections=self.butler.collections)
+
+    def getVisitDataId(self, dataId):
+        """Get the visitId for the dataId.
+        """
+        expId = butlerUtils.getExpIdFromDayObsSeqNum(self.butler, dataId)
+        visitDataIds = self.butler.registry.queryDataIds(["visit", "detector"], dataId=expId)
+        visitDataIds = list(set(visitDataIds))
+        if len(visitDataIds) == 1:
+            visitDataId = visitDataIds[0]
+            return visitDataId
+        else:
+            self.log.warning(f"Failed to find visitId for {dataId}, got {visitDataIds}. Do you need to run"
+                             " define-visits?")
+            return None
+
+    def safeWrite(self, object, datasetType, dataId):
```

Comment:
I think this function needs to be renamed: given that it actually can remove existing data this doesn't seem like the best words for it. `overwriteIfExisting` or something like that?

Reply:
how about just `clobber`